src/mds/populate.py

```diff
@@ -117,11 +82,35 @@ def normalize(entry: Dict[Any, Any]):
 
         data = [normalize(x) for x in mds_arr]
 
-        # build index of keys. which is used to compute the index into the .metadata array
-        # Admittedly a hack but will be faster than using json path, until the release of RedisJson v1.2
         keys = list(results.keys())
         info = {"commons_url": common.commons_url}
-        await datastore.update_metadata(name, mds_arr, keys, tags, info, aggregations)
+        res = await datastore.update_metadata(name, mds_arr, keys, tags, info)
+
+
+async def main(commons_config: Commons, hostname: str, port: int) -> None:
+    """
+    Given a config structure, pull all metadata from each one in the config and cache into the following
+    structure:
+    {
+       "commons_name" : {
+                    "metadata" : [ array of metadata entries ],
+                    "field_mapping" : { dictionary of field_name to column_name },
+                    "guids: [ array of guids, used to index into the metadata array ],
+                    "tags": { 'category' : [ values ] },
+                    "commons_url" : "url of commons portal"
+        },
+        "..." : {
+        }
+    """
+
+    if not config.USE_AGG_MDS:
+        print("aggregate MDS disabled")
+        exit(1)
+
+    await datastore.init(hostname, port)
+    await datastore.drop_all()
```

Comment:
Great points. I punted on that rather than doing the work to diff/incrementally update records. At this point dropping/re-populating is definitely easier, but I'll spend some time to look at an incremental diff. One way or another I'm sure we're going to need that in time and better to at least get some insight on that before we desperately need it.